sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/spice_indexer.py

```diff
@@ -0,0 +1,369 @@
+            ]
+        )
+
+    return results_j2000, results_datetime, results_sclk
+
+
+def _upsert_into_spice_table(
+    spice_object: SPICEFilePath,
+    file_coverage_j2000: list[list[float]],
+    file_coverage_datetime: list[list[datetime]],
+    file_coverage_sclk: list[list[str]],
+    latest_sclk: Path,
+    latest_lsk: Path,
+):
+    """Insert/Update the spice metadata table with collected data.
+
+    Parameters
+    ----------
+    spice_object: SPICEFilePath
+        The SPICE file to upsert
+    file_coverage_j2000: list[list[float]]
+        A list of file intervals in j2000 time format
+    file_coverage_datetime: list[list[datetime]]
+        A list of file intervals in datetime format
+    file_coverage_sclk: list[list[str]]
+        A list of file intervals in sclk string format
+    latest_sclk: Path
+        The latest clock kernel used for the above calculations
+    latest_lsk: Path
+        The latest leapsecond kernel used for the above calculations
+    """
+    # Format the data to insert
+    spice_params = {}
+    spice_params["ingestion_date"] = datetime.now()
+    spice_params["kernel_type"] = spice_object.spice_metadata["type"]
+    spice_params["version"] = spice_object.spice_metadata["version"]
+    spice_params["file_name"] = str(spice_object.filename.name)
+    spice_params["file_root"] = str(spice_object.filename.name).replace(
+        str(spice_object.spice_metadata["version"]), ""
+    )
+    spice_params["min_date_j2000"] = file_coverage_j2000[0][0]
+    spice_params["max_date_j2000"] = file_coverage_j2000[-1][-1]
+    spice_params["file_intervals_j2000"] = file_coverage_j2000
+    spice_params["min_date_datetime"] = file_coverage_datetime[0][0]
+    spice_params["max_date_datetime"] = file_coverage_datetime[-1][-1]
+    spice_params["file_intervals_datetime"] = [
+        [dt.isoformat() for dt in sublist] for sublist in file_coverage_datetime
+    ]
+    spice_params["min_date_sclk"] = file_coverage_sclk[0][0]
+    spice_params["max_date_sclk"] = file_coverage_sclk[-1][-1]
+    spice_params["file_intervals_sclk"] = file_coverage_sclk
+    spice_params["lsk_kernel"] = str(latest_lsk)
+    spice_params["sclk_kernel"] = str(latest_sclk)
+
+    # Open db session and determine whether to update the database
+    # or insert a new record
+    with db.Session() as session, session.begin():
+        # Check if the record already exists
+        existing_entry = (
+            session.query(models.SPICEFiles)
+            .filter_by(file_name=spice_params["file_name"])
+            .first()
+        )
+        if existing_entry:
+            for key, value in spice_params.items():
+                if key == "file_name":
+                    continue
+                setattr(existing_entry, key, value)  # Update existing record
+        else:
+            # Record does not exist, add it
+            session.add(models.SPICEFiles(**spice_params))
+    logger.info("Wrote data to the SPICEFiles table")
```

Comment:
log what you added for helpful future debugging?
```suggestion
    logger.info(f"Wrote {spice_params} to the SPICEFiles table")
```